File: remote-query/src/main/scala/org/littlewings/infinispan/query/Book.scala
package org.littlewings.infinispan.query

object Book {
  def apply(isbn: String, title: String, price: Int, summary: String): Book = {
    val book = new Book
    book.isbn = isbn
    book.title = title
    book.price = price
    book.summary = summary
    book
  }
}

class Book {
  var isbn: String = _
  var title: String = _
  var price: Int = _
  var summary: String = _
}

File: remote-testcontainers/src/main/java/org/littlewings/infinispan/remote/testcontainers/Book.java
package org.littlewings.infinispan.remote.testcontainers;

import org.infinispan.protostream.annotations.ProtoFactory;
import org.infinispan.protostream.annotations.ProtoField;

public class Book {
    @ProtoField(number = 1)
    String isbn;

    @ProtoField(number = 2)
    String title;

    @ProtoField(number = 3, defaultValue = "0")
    int price;

    @ProtoFactory
    public static Book create(String isbn, String title, int price) {
        Book book = new Book();

        book.isbn = isbn;
        book.title = title;
        book.price = price;

        return book;
    }

    public String getIsbn() {
        return isbn;
    }

    public String getTitle() {
        return title;
    }

    public int getPrice() {
        return price;
    }
}
